src/core/lib/surface/call.c

```diff
@@ -1461,6 +1461,9 @@ static grpc_call_error call_start_batch(grpc_exec_ctx *exec_ctx,
         grpc_slice_buffer_stream_init(
             &call->sending_stream,
             &op->data.send_message->data.raw.slice_buffer, op->flags);
+        if (op->data.send_message->data.raw.compression > GRPC_COMPRESS_NONE) {
```

Comment:
Can you add a comment before this block explaining what's going on? For example:
```
/* If the outgoing buffer is already compressed, mark it as so in the flags. 
These will be picked up by the compression filter and further (wasteful) attempts at compression skipped */
```